controllers/helmchart_controller.go

```diff
@@ -550,7 +550,8 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 
 		// Tell the chart repository to use the OCI client with the configured getter
 		clientOpts = append(clientOpts, helmgetter.WithRegistryClient(registryClient))
-		ociChartRepo, err := repository.NewOCIChartRepository(repo.Spec.URL, repository.WithOCIGetter(r.Getters), repository.WithOCIGetterOptions(clientOpts), repository.WithOCIRegistryClient(registryClient))
+		url := strings.TrimSuffix(repo.Spec.URL, "/")
```

Comment:
Maybe we should use the normalized URL throughout this whole function as the `helmgetter`, `tlsConfig`, `loginOpt` are still configured with the original URL. Maybe move it to the beginning of the function.
Also rename the variable accordingly, like `normalizedURL`.